web/src/sidebar_ui.ts

```diff
@@ -53,6 +53,25 @@ export function update_invite_user_option(): void {
     }
 }
 
+export function update_left_sidebar_unread_counts_visibility(): void {
+    const $home_view = $(".selected-home-view");
+    const $topics_header = $("#topics_header");
+    const $streams_header = $("#streams_header");
+    const $direct_message_header = $("#private_messages_section_header");
+
+    if (!user_settings.web_left_sidebar_unreads_count_summary) {
+        $home_view.addClass("hide_unread_counts");
```

Comment:
Maybe call this class `hide-unread-count-summary` -- that aligns a lot more closely with the setting name. And we're preferring `-`s in new CSS class names and IDs there days.